### PR TITLE
Addition of glossary

### PR DESCRIPTION
**Pull request by sbtaylor15**
## Description

This PR fixes the last remaining items on PR #755.  This PR replaces #755.  Also, this is a first pass at the Glossary.

Fixes pyrsia/pyrsia#

This PR does... by accomplishing... and it can be reviewed by... you can also test the changes by running...

## PR Checklist

- [X] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [X] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [X] I've included a good title and brief description along with how to review them.
- [X] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [X] I've built the code `cargo build --all-targets` successfully.
- [X] I've run the unit tests `cargo test --workspace` and everything passes.
- [X] I've made sure my rust toolchain is current `rustup update`.

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+This will be a collection of terms and their meanings that have been widely used across the projects. It will be a good reference point for the developers to understand and comprehend the components of the Pyrisa Project.
+
+## Terms
+
+- Block: A collection of cryptographically verifiable transactions and a header providing provenance
+  - Finalized Block: Blocks which have been agreed upon by the Network
+- Node: a computer participating in the blockchain
+- Authoritative node: A node possessing and signing transactions using a trusted key - where
+    "trusted key" is defined as being a key that is stored on the block chain and authorized.
```
Comment (tiainen) [inline docs/glossary.md:11]: For consistency?
```suggestion
    "trusted key" is defined as being a key that is stored on the blockchain and authorized.
```

File: docs/glossary.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- Proof of Authority: One of many methods of consensus used in Blockchain.
+    Proof of Authority accepts that there are a limited number of participants in the network that can commit new transactions to the blockchain (See: Authoritative node). Transactions in a block, and the block it self, must all be signed by an authority certificate.
+- Provenance logs: A chain of cryptographically verified information establishing the provenance (origin) of a given package.
+- Blockchain: An append only, distributed, data-store that provides cryptographic provability of all transactions
+    Search index: Generally refers to any indexing tech that is used to in search tech to quickly look up results: See "inverse index"
```
Comment (tiainen) [inline docs/glossary.md:16]: The _See "inverse index"_ begs for a link to find out more information? Either remove it or provide more information.

File: docs/glossary.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+## Terms
+
+- Block: A collection of cryptographically verifiable transactions and a header providing provenance
+  - Finalized Block: Blocks which have been agreed upon by the Network
+- Node: a computer participating in the blockchain
+- Authoritative node: A node possessing and signing transactions using a trusted key - where
+    "trusted key" is defined as being a key that is stored on the block chain and authorized.
+- Proof of Authority: One of many methods of consensus used in Blockchain.
+    Proof of Authority accepts that there are a limited number of participants in the network that can commit new transactions to the blockchain (See: Authoritative node). Transactions in a block, and the block it self, must all be signed by an authority certificate.
```
Comment (tiainen) [inline docs/glossary.md:13]:
```suggestion
    Proof of Authority accepts that there are a limited number of participants in the network that can commit new transactions to the blockchain (See: Authoritative node). Transactions in a block, and the block itself, must all be signed by an authority certificate.
```

File: docs/glossary.md (hunk 2 above)
Comment (sbtaylor15) [reply docs/glossary.md]: Removed the See "inverse index"

**Review by tiainen — APPROVED** [review]

**Pull request merged**